### DISCUSSION TITLE
Custom DefaultShortUrl

### DISCUSSION THREAD

**Discussion by Z3d0X** [ash-jc-allen/short-url#125] [Ideas]
Looking through the docs & source code, i found no way to customize the DefaultShortUrl.
I would like to be able to change the `$baseUrl` instead of the `config('app.url')`
```php
/**
 * Build and return the default short URL that will be stored in the
 * database.
 *
 * @return string
 */
private function buildDefaultShortUrl(): string
{
    $baseUrl = config('app.url').'/';

    if ($this->prefix() !== null) {
        $baseUrl .= $this->prefix().'/';
    }

    return $baseUrl.$this->urlKey;
}
```

**Comment by ash-jc-allen** [comment]
Hey @ZedoXmv! 😄

At the moment, there isn't any way in the package to change the domain used for the default short URL.

If you'd like to use a different domain with your short URLs, you could use a custom route to do this (https://github.com/ash-jc-allen/short-url#custom-route). Or, I'd be happy to review a pull request if you'd like to make one

**Reply by Z3d0X** [reply]
thanks ill use a custom route for now, maybe ill take a shot a PR later! cheers 😄 

leaving my implementation here for any who's looking for this
```php
public function handle(string $destination_url): string
{
    $shortUrl = ShortURL::destinationUrl($destination_url)->make();
    $key = $shortUrl->url_key;

    return route('short-url', $key); //custom route
}
```
